lib/signer.groovy

```diff
@@ -0,0 +1,46 @@
+#!./lib/runner.groovy
+import net.sf.json.*
+import org.jvnet.hudson.update_center.Signer
+import groovy.io.FileType
+
+def list = []
+
+def dir = new File('./target/')
+dir.eachFileRecurse (FileType.FILES) { file ->
+  if(file.getPath().endsWith('.json'))
+    list << file;
+}
+
+list.each {
+  String file  = it.path
+  println "== ${file}"
+
+  // Load content of JSON file to be signed
+  File inputfile = new File(file)
+  String utf8Content = inputfile.getText("UTF-8")
+  if(utf8Content.length()<1) {
```

Comment:
`[]` is a really dangerous file to sign. Not strictly empty, but close enough. So this doesn't help.

Would we ever want to sign something with less than 100 or 200 bytes?

Reply:
Maybe too naive but how about testing the presence of `":` in utf8Content?

Reply:
ok, I've added 2 rules: it must contains a `: ` (as per @lemeurherve ) and the size should be more than 100 chars (as per @daniel-beck 's feedback).

Tested on the actual JSON files and it works
Tested with `[]`, `{}` (with and without JSONP) and fails as expected.